rpc.go

```diff
@@ -25,17 +25,36 @@ type Rpc struct {
 
 type opts struct {
 	cors      string
+	headers   string
+	methods   string
 	maxMemory int64
 }
 
 type OptsFunc func(*opts)
 
+// Deprecated: please use WithParams
 func WithCors(allowedOrigins string) OptsFunc {
 	return func(opts *opts) {
 		opts.cors = allowedOrigins
 	}
 }
 
+func WithParams(cors, headers, methods []string) OptsFunc {
```

Comment:
It's unclear name, it's better to use `WithCors` or `WithCors2` to save back compatibility. `WithCors` should call `WithCors2` with default arguments.